### PR TITLE
refactoring: make CircuitTransformer into an abstract interface

### PR DESCRIPTION
**Pull request by weinstein**
this way as additional transformations are implemented (see #127 #132 #133) we can implement additional transformations and swap them out more easily (ex for evaluation).

### CODE REVIEW

**Review by dstrain115 — APPROVED** [review]

File: recirq/quantum_chess/circuit_transformer.py [hunk 1]
```diff
@@ -22,19 +22,31 @@
 
 
 class CircuitTransformer:
+    """Abstract interface for circuit transformations.
+
+    For example: named -> grid qubit transformations.
```
Comment (wingers) [inline recirq/quantum_chess/circuit_transformer.py:27]: nit: NamedQubit -> GridQubit transformations

File: recirq/quantum_chess/circuit_transformer.py [hunk 2]
```diff
@@ -22,19 +22,31 @@
...
+    """
+    def __init__(self):
```
Comment (wingers) [inline recirq/quantum_chess/circuit_transformer.py:29]: nit: should we add `device` as a parameter?

**Review by wingers — APPROVED** [review]

File: recirq/quantum_chess/circuit_transformer.py (hunk 2 above)
Comment (weinstein) [reply recirq/quantum_chess/circuit_transformer.py:29]: IMO there's not much benefit to have the base class take the device and do nothing else with it besides storing a data member, even if all the implementations will end up doing this. If the implementations end up sharing some common functionality, there will probably be a cleaner way to share that.

Comment (wingers) [reply recirq/quantum_chess/circuit_transformer.py:29]: sgtm 👍

File: recirq/qaoa/placement.py [hunk 3]
```diff
@@ -103,7 +103,7 @@ def place_on_device(circuit: cirq.Circuit,
                      for n1, n2 in unit.initial_map.items()}
     final_map = {tk_to_cirq_qubit(n1): tk_to_cirq_qubit(n2)
              for n1, n2 in unit.final_map.items()}
-    routed_circuit = pytket.cirq.tk_to_cirq(unit.circuit)
+    routed_circuit = pytket.extensions.cirq.tk_to_cirq(unit.circuit)
```
Comment (dstrain115) [inline recirq/qaoa/placement.py:106]: Oh, I saw this on another PR, Nice job tracking this down!

**Review by dstrain115 — APPROVED** [review]

File: recirq/qaoa/placement.py (hunk 3 above)
Comment (mpharrigan) [reply recirq/qaoa/placement.py:106]: ah in the future can you ping me or open this as a separate fix PR. This issue also exists in the notebooks and I had to re-discover the solution xref #148

Comment (weinstein) [reply recirq/qaoa/placement.py:106]: my bad! Will do next time.

Comment (mpharrigan) [reply recirq/qaoa/placement.py:106]: no worries! I was trying to complain at @dstrain115  :)

**Pull request merged**